src/pages/AllCoins/index.js

```diff
@@ -1,12 +1,204 @@
+
+            <CoinContainer theme={this.props.theme}>
+                <Row>
+                  {this.widths.map((width, index) => <Cell key={keyGen()} width={width} weight={this.fontWeightBold} size={this.labelsFontSize}>{this.labels[index]}</Cell>)}
+                </Row>
+              {data.map((coin, index) => {
+                const widths = this.widths;
+                const colors = this.colors;
+                const { 
+                  circulating_supply,
+                  current_price,
+                  image,
+                  market_cap,
+                  name,
+                  symbol,
+                  total_supply,
+                  total_volume,
+                  price_change_percentage_1h_in_currency: hourChange, 
+                  price_change_percentage_24h_in_currency: twentyFourHourChange, 
+                  price_change_percentage_7d_in_currency: sevenDayChange, 
+                  sparkline_in_7d: sevenDayPriceList
+                  } = coin;
+                return (
+                <Row key={keyGen()}>
+
+                  <Cell key={keyGen()} width={widths[0]}>
+                    {index + 1}
+                  </Cell>
+                  
+                  <Cell key={keyGen()} width={widths[1]}>
+                    <Img src={image} alt={name} />
+                    {name} ({symbol.toUpperCase()})
+                  </Cell>
+                  
+                  <Cell key={keyGen()} width={widths[2]} >
+                    {convertLargeNumber(formatCurrency(current_price, this.props.currency, "en"))}
+                  </Cell>
+                  
+                  <Cell key={keyGen()} width={widths[3]} number={hourChange}>
+                    {(this.props.currency !== symbol && (
+                        <>
+                          <Arrow content={getArrow(hourChange)} />
+                          {Math.abs(hourChange.toFixed(2))}% 
+                        </>
+                      )) || <span>-</span>
+                    }
+                  </Cell>
+                  
+                  <Cell key={keyGen()} width={widths[4]} number={twentyFourHourChange}> 
+                    {(this.props.currency !== coin.symbol && (
+                        <>
+                          <Arrow content={getArrow(twentyFourHourChange)} />
+                          {Math.abs(twentyFourHourChange.toFixed(2))}%
+                        </>
+                      )) || <span>-</span>
+                    }
+                  </Cell>
+                  
+                  <Cell key={keyGen()} width={widths[5]} number={sevenDayChange}>
+                    {(this.props.currency !== symbol && (
+                        <>
+                          <Arrow content={getArrow(sevenDayChange)} />
+                          {Math.abs(sevenDayChange.toFixed(2))}%
+                        </>
+                      )) || <span>-</span>
+                    }                    
+                  </Cell>

```

Comment:
can you make some of these individual components?

Reply:
Done. Refactored this and entire project with more components for readability.